R/population-study-design.R

```diff
@@ -0,0 +1,226 @@
+#' @title addStudyParameters
+#' @param population `Population` object
+#' @param studyDesignFile file name of study design table
+#' @export
+#' @import ospsuite
+addStudyParameters <- function(population, studyDesignFile = NULL) {
+  validateIsOfType(population, "Population")
+  validateIsString(studyDesignFile, nullAllowed = TRUE)
+  if (is.null(studyDesignFile)) {
+    return(invisible())
+  }
+  studyDesign <- loadStudyDesign(studyDesignFile)
+
+  initialTargetValues <- rep(NA, population$count)
+  populationData <- ospsuite::populationAsDataFrame(population)
+
+  for (target in studyDesign$targets) {
+    parameterPath <- target$name
+
+    updatedTargetValues <- updateTargetValues(initialTargetValues, target$values, studyDesign$source, populationData)
+
+    population$setParameterValues(parameterPath, updatedTargetValues)
+  }
+}
+
+#' @title updateTargetValues
+#' @param values initial vector to update
+#' @param targetValues vector of values to be assigned using `sourceExpressions`
+#' @param sourceExpressions study design expressions to be evaluated
+#' @param data population data as data.frame
+#' @export

```

Comment:
Is this really a public method to export ? This looks like internal stuff to me

Reply:
You are right, there are a couple others I should not export.
I tend to do this at the beginning to test the functions works correctly when updating the package. I'll remove them asap.